File: backend/app/schemas/vehicle.py
from pydantic import BaseModel
from typing import Optional


class VehicleOut(BaseModel):
    id: int
    asset_id: str
    asset_type: Optional[str] = None
    trailer_type: Optional[str] = None


class VehicleSummaryOut(BaseModel):
    vehicle_id: int
    asset_id: Optional[str] = None
    asset_type: Optional[str] = None
    total_trips: int = 0
    drivers_used: int = 0
    avg_speed_kmph: Optional[float] = None
    total_distance_km: Optional[float] = None
    avg_distance_km: Optional[float] = None
    eta_success_rate: Optional[float] = None

File: frontend/src/types/vehicle.ts
export interface VehicleSummaryRow {
  vehicle_id: number;
  asset_id: string;
  asset_type: string;
  total_trips: number;
  drivers_used: number;
  avg_speed_kmph: number;
  total_distance_km: number;
  avg_distance_km: number;
  eta_success_rate: number;
}

export interface VehicleDriver {
  driver_id: number;
  driver_name: string;
  trip_count: number;
}

export interface VehicleDetail {
  summary: VehicleSummaryRow;
  drivers_used: VehicleDriver[];
  recent_trips: Array<{
    id: number;
    dispatch_entry_no: string;
    driver_name: string;
    origin_name: string;
    destination_name: string;
    trip_start: string;
    trip_duration_minutes: number;
    eta_met: boolean;
    avg_speed_kmph: number;
  }>;
}
